test/api/repository/survey_repository_test.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/repository/survey_repository.dart';
+import 'package:flutter_survey/api/response/base/base_http_response.dart';
+import 'package:flutter_survey/api/response/survey_response.dart';
+import 'package:flutter_survey/models/survey.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../mock/mock_data.mocks.dart';
+import '../../utils/file_util.dart';
+
+void main() {
+  group("SurveyRepositoryTest", () {
+    late MockSurveyService mockSurveyService;
+    late SurveyRepository surveyRepository;
+    late Map<String, dynamic> surveysJson;
+
+    setUp(() async {
+      mockSurveyService = MockSurveyService();
+      surveyRepository = SurveyRepositoryImpl(mockSurveyService);
+      surveysJson = await FileUtil.loadFile('test_resources/surveys.json');
+    });
+
+    test(
+        'When calling getSurveys successfully, it returns corresponding response',
+        () async {
+      final surveyResponses =
+          BaseHttpResponseList<SurveyResponse>.fromJson(surveysJson);
+      final surveys = surveyResponses.data
+          .map((response) => response.attributes)
+          .toList()
+          .map((response) => response.toSurvey())
+          .toList();
```

Comment:
lol are you writing real code in test? 😂  I think for the test, after receiving the map `data` you can do direct access response['data']['survey']['id'] to verify the output quickly yeah?

Reply:
ok, I decided to switch my test assertion to assert the result list size and 2 first titles 😄 
```
      expect(result.length, 2);
      expect(result[0].title, "Scarlett Bangkok");
      expect(result[1].title, "ibis Bangkok Riverside");
```

Reply:
that should be enough 👍